refinery/core/api.py

```diff
@@ -678,24 +671,34 @@ def filter_by_group(self, request, obj_list):
         return obj_list
 
     def obj_get(self, bundle, **kwargs):
-        dataset = SharableResourceAPIInterface.obj_get(self, bundle, **kwargs)
-        return self.handle_incomplete_dataset(dataset)
+        try:
+            dataset = DataSet.objects.get(uuid=kwargs["uuid"])
+        except (DataSet.DoesNotExist, DataSet.MultipleObjectsReturned) as e:
+            logger.error("Couldn't properly fetch DataSet with UUID: %s %s",
+                         kwargs["uuid"], e)
+            raise NotFound(e)
+
+        if not dataset.is_valid():
+            raise NotFound(
+                "DataSet with UUID: {} is invalid, and most likely is "
+                "still being created".format(dataset.uuid)
+            )
+        return SharableResourceAPIInterface.obj_get(self, bundle, **kwargs)
 
     def obj_get_list(self, bundle, **kwargs):
-        datasets = SharableResourceAPIInterface.obj_get_list(
-            self,
-            bundle,
-            **kwargs
-        )
-        complete_datasets = []
+        datasets = SharableResourceAPIInterface.obj_get_list(self, bundle,
```

Comment:
probably need to call `get_obj_list()` of `ModelResource` and remove invalid data sets from the bundle then
```python
datasets = super(SharableResourceAPIInterface, self).obj_get_list(bundle, **kwargs)
```